bus-mapping/src/evm/opcodes/sload.rs

```diff
@@ -0,0 +1,182 @@
+use super::Opcode;
+use crate::{
+    exec_trace::ExecutionStep,
+    operation::{container::OperationContainer, StackOp, StorageOp, RW},
+    Error,
+};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it corresponding to the
+/// [`OpcodeId::SLOAD`](crate::evm::OpcodeId::SLOAD) `OpcodeId`.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Sload;
+
+impl Opcode for Sload {
+    #[allow(unused_variables)]
+    fn gen_associated_ops(
+        &self,
+        exec_step: &mut ExecutionStep,
+        container: &mut OperationContainer,
+        next_steps: &[ExecutionStep],
+    ) -> Result<usize, Error> {
+        let gc_idx_start = exec_step.gc().0;
+        let mut gc_idx = gc_idx_start;
+
+        // First stack read
+        let stack_value_read = exec_step
+            .stack()
+            .last()
+            .cloned()
+            .ok_or(Error::InvalidStackPointer)?;
+        let stack_position = exec_step.stack().last_filled();
+
+        // Manage first stack read at latest stack position
+        gc_idx += 1;
+        exec_step.bus_mapping_instance_mut().push(container.insert(
```

Comment:
As you suggested. We can maybe add a function that wraps both actions at the same time :smile: